### PR TITLE
Fixed bug related to missing_value for android and linux

### LINKED ISSUE #1879: Android table missing_value
### Describe the bug

Missing_value in Table does not work on Android as intended.

### Steps to reproduce

Step1: Use table in your project
Step2: Use more columns in your table than is in your data
Step3: get error            04-17 08:55:26.553  3636  3636 E AndroidRuntime: Caused by: com.chaquo.python.PyException: AttributeError: 'Row' object has no attribute 'short'   

#this throws error on Android, but runs on Windows
self.data = [("root%s" % i, "value %s" % i) for i in range(1, 10)]
#this does throw error and runs on Android too
#self.data = [("root%s" % i, "value %s" % i, "short %s" % i) for i in range(1, 10)]

self.left_container = toga.Table(headings=["source", "number", "short"], data=self.data, on_select=self.get_file_content_and_display, missing_value="")

### Expected behavior

App won't throw error and will fill missing values with blank string.

### Screenshots

_No response_

### Environment

- Operating System: Windows 10
- Python version: 3.10.10
- Software versions: 
  - Briefcase: 0.3.14
  - Toga: 0.3.1

### Logs

```

```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by LunaMeadows**
For linux, I passed the missing_value from self.interface.missing_value from Tree to the sourcetreemodel. I did this because self.instance is not available inside the sourcetreemodel but the value is needed to be able to solve the bug.

For android, I simply added the self.interface.missing_value as the default option for the getattr.

On android the app will crash if values are not provided for all headings, on linux it will cause an error in the logs. 

Fixes #1879 

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: gtk/src/toga_gtk/widgets/tree.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ def create(self):
         self.store = SourceTreeModel(
             [{"type": str, "attr": a} for a in self.interface._accessors],
             is_tree=is_tree,
+            missing_value=getattr(self.interface, "missing_value", None),
```
Comment (freakboy3742) [inline gtk/src/toga_gtk/widgets/tree.py:18]: Why use getattr here? Unless I'm missing something, `missing_value` should always be defined on the interface.

Comment (LunaMeadows) [reply gtk/src/toga_gtk/widgets/tree.py:18]: I am not at my computer to confirm fully but I believe the issue was the Tree interface did not have the missing_value. If you are able to look at the first round of checks it should show which interface did have it

Comment (freakboy3742) [reply gtk/src/toga_gtk/widgets/tree.py]: Ah - that makes sense. That's an API inconsistency that should probably be resolved...

**Review by freakboy3742 — APPROVED** [review]
The missing_value handling on tree is something we need to backfill for tree, but otherwise, this looks great! Thanks!

**Pull request merged**